lib/graphql/with-ssr-query.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { QueryOptions } from '@apollo/client';
+import { Query } from '@apollo/client/react/components';
+import { getDataFromTree } from '@apollo/client/react/ssr';
+import { cloneDeep, isUndefined, omitBy, pick } from 'lodash';
+
+import { APOLLO_STATE_PROP_NAME, APOLLO_VARIABLES_PROP_NAME, initClient } from '../apollo-client';
+
+type QueryParams = QueryOptions & {
+  getVariablesFromProps?: (props: any) => any;
+  /** Whether to fetch data for nested queries. This currently doesn't work for pages with a `useRouter` call, see https://github.com/opencollective/opencollective/issues/7013 */
+  fetchDataFromTree?: boolean;
+  /** @deprecated Whether to use the legacy data structure from Apollo, as returned by `withData`. Intended to make the transition from old pages smoother, should not be used in new developments */
+  useLegacyDataStructure?: boolean;
+  /** A custom function to preload more data on the client */
+  preload?: (client: any, mainQueryResult: any) => Promise<void>;
+};
+
+export const ssrGraphQLQuery = ({
```

Comment:
I went with `ssrGraphQLQuery` because the signature almost matches and is meant to replace the one of `query` (from `apollo`). A more standard name I can suggest is ` withSSRGraphQLQuery`, wdyt?